File: app/src/main/java/br/com/jonathanzanella/myexpenses/source/SourceContract.java
package br.com.jonathanzanella.myexpenses.source;

import android.support.annotation.StringRes;

import br.com.jonathanzanella.myexpenses.validations.ValidationError;

/**
 * Created by jzanella on 8/27/16.
 */

interface SourceContract {
	interface View {
		void setTitle(@StringRes int string);
		void showSource(Source source);
		Source fillSource(Source source);
		void finishView();
		void showError(ValidationError error);
	}
}

File: app/src/main/kotlin/br/com/jonathanzanella/myexpenses/source/SourceContract.kt
package br.com.jonathanzanella.myexpenses.source

import android.content.Context
import android.support.annotation.StringRes

import br.com.jonathanzanella.myexpenses.validations.ValidationError

interface SourceContract {
    interface View {
        val context: Context
        fun setTitle(@StringRes string: Int)
        fun setTitle(string: String)
        fun showSource(source: Source)
    }

    interface EditView : View {
        fun fillSource(source: Source): Source
        fun finishView()
        fun showError(error: ValidationError)
    }
}
